### ISSUE TITLE
LDAP-197: Annotation Support for Property Mapping

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-ldap#240]
[Jamie Goodfellow](https://jira.spring.io/secure/ViewProfile.jspa?name=jgoodfel) (Migrated from [LDAP-197](https://jira.spring.io/browse/LDAP-197?redirect=false)) said:

I would like to be able to put annotations on my domain objects to define what LDAP properties map to POJO fields.  Perhaps there could also be a class-level annotation that defines a POJO as an LDAP POJO.  Using this class-level annotation, POJO fields could be mapped by default to LDAP properties of the same name, but field annotations could override.

Basically this system could use the same mapping strategy as JPA annotations for DBs.
